syntax = "proto3";

message Request {
  string id = 1;
  string bench_type = 2;
  string metric = 3;
  int32 batch_id = 5;
  int32 batch_unit = 4;
  int32 batch_size = 6;
  string analysis_parameter = 7;
}

message Response {
  string id = 1;
  int32 last_batch_id = 2;
  repeated Batch batches = 3;
  double analysis =4;
}
message Batch {
  repeated float samples = 1;
}
